packages/core/ui/styling/converters.ts

```diff
@@ -9,8 +9,25 @@ const STYLE_CURVE_MAP = Object.freeze({
 	spring: CoreTypes.AnimationCurve.spring,
 });
 
+/**
+ * Converts a time value string to milliseconds.
+ * Supports both 's' (seconds) and 'ms' (milliseconds) units.
+ * @param value The time value as a string (e.g., '1s', '500ms', '2.5')
+ * @returns Time in milliseconds, always >= 0
+ * @throws Error if the input is not a valid time value
+ */
 export function timeConverter(value: string): number {
-	let result = parseFloat(value);
+	if (!value || typeof value !== 'string') {
+		throw new Error(`Invalid time value: '${value}'. Expected a string with time value.`);
+	}
+
+	const numericValue = parseFloat(value);
+	if (isNaN(numericValue)) {
+		throw new Error(`Invalid time value: '${value}'. Expected a numeric value with optional 's' or 'ms' unit.`);
```

Comment:
```suggestion
		console.warn(`Invalid time value: '${value}'. Expected a numeric value with optional 's' or 'ms' unit.`);
```

```suggestion
		console.warn(`Invalid time value: '${value}'. Expected a string with time value.`);
```

we avoid unnecessary `throw` in core.